<!-- <launch>
    <node pkg="robot_controller" type="test.py" name="Test" output="screen"/>
</launch> -->
<launch>
    <arg name="robot_namespace" default="/robot_0"/>
    <arg name="lidar_port" default="/dev/lidar"/>
    <arg name="imu_port" default="/dev/imu"/>
    <arg name="laser_scan_topic_name" default="$(arg robot_namespace)/scan"/>
    <!-- replace with ros master ip and port -->
    <!-- <env name="ROS_MASTER_URI" value="http://192.168.1.100:11311"/> -->
    <!-- replace with current pc ip -->
    <!-- <env name="ROS_HOSTNAME" value="192.168.1.101"/>  -->

    <!-- <include file="$(find robot_controller)/launch/robot_bringup.launch" /> -->
    <!-- Start lidar node -->
    <include file="$(find ldlidar_ros)/launch/ld19.launch">
        <arg name="port_name" value="$(arg lidar_port)"/>
        <arg name="laser_scan_topic_name" value="$(arg laser_scan_topic_name)"/>
        <!-- <remap from="/scan" to="$(arg robot_namespace)/scan"/> -->
    </include>
    <!-- Start IMU node -->
    <node pkg="wit_ros_imu" type="wit_normal_ros.py" name="imu" output="screen">
        <param name="port"               type = "str"    value="$(arg imu_port)"/>
        <param name="baud"              type = "int"    value="9600"/>
        <param name="frame_id"          value="dummy_base_link"/>
	    <remap from="/wit/imu" to="$(arg robot_namespace)/imu/data"/>
	    <remap from="/wit/yaw" to="$(arg robot_namespace)/imu/yaw"/>
    </node>

    
    <!-- Start robot_counter_service -->
    <node pkg="robot_controller" type="robot_counter_service.py" name="robot_counter_service" output="screen"/>
    <!-- Start robot_0_model -->
    <!-- <node pkg="robot_controller" type="run_model.py" name="model" ns="$(arg robot_namespace)" output="screen"/> -->
    <!-- Start robot_0_execute -->
    <node pkg="robot_controller" type="run_robot.py" name="execute" ns="$(arg robot_namespace)" output="screen">
	    <!-- <remap from="/odom" to="$(arg robot_namespace)/odom"/> -->
    </node>

    <!-- Start robot_localization with UKF fusion -->
    <node pkg="robot_localization" type="ukf_localization_node" name="ukf_localization_node" output="screen">
        <rosparam command="load" file="$(find robot_controller)/config/ukf.yml"/>
        <param name="robot_namespace" value="$(arg robot_namespace)"/>
	   <remap from="/odometry/filtered" to="$(arg robot_namespace)/odometry/filtered"/>
    </node>

    <!-- Topic remappings -->
    <!-- <node pkg="topic_tools" type="relay" name="relay_cmd_vel" args="/cmd_vel $(arg robot_namespace)/cmd_vel" output="screen"/> -->
    <!-- <node pkg="topic_tools" type="relay" name="relay_odom" args="/odom $(arg robot_namespace)/odom" output="screen"/> -->
    <!-- <node pkg="topic_tools" type="relay" name="relay_imu" args="/imu $(arg robot_namespace)/imu" output="screen"/> -->
    <!-- <node pkg="topic_tools" type="relay" name="relay_odometry_filtered" args="/odometry/filtered $(arg robot_namespace)/odometry/filtered" output="screen"/> -->
    <!-- <node pkg="topic_tools" type="relay" name="relay_scan" args="/scan $(arg robot_namespace)/scan" output="screen"/> -->
    
    
    <!-- Reserve for Rviz -->

    <!-- ============================================== -->
    <!-- ROBOT DESCRIPTION & TF -->
    <!-- ============================================== -->

    <!-- Load robot URDF -->
    <param name="robot_description" command="cat $(find robot_description)/urdf/robot_description.urdf" />

    <!-- Robot State Publisher - publishes TF from URDF -->
    <!-- TF chain: dummy_base_link -> base_link -> lidar -->
    <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
        <param name="publish_frequency" value="10.0"/>
    </node>

</launch>
